q2_dsfdr/plugin_setup.py

```diff
@@ -40,16 +40,24 @@ def permutation_fdr(output_dir: str,
         index_fp = os.path.join(output_dir, 'index.html')
 
         metadata_series = metadata.to_series()[table.index]
-        uvals = metadata_series.unique()
-        if len(uvals) < 2:
-            raise ValueError('Only one value in mapping file data column (%s). Aborting' % uvals[0])
-        if len(uvals) > 2 and statistical_test != 'kruwallis':
-            raise ValueError('More than two values in mapping file data column (%s). Aborting' % uvals)
 
-        # convert labels into incremental integers
-        labels = np.zeros(len(metadata_series))
-        for i in range(1, len(uvals)):
-            labels[metadata_series == uvals[i]] = i
+        # numeric metadata
+        if metadata.type == 'numeric':
+            if statistical_test not in ('spearman', 'pearson', 'nonzerospearman', 'nonzeropearson'):
+                raise ValueError('Cannot perform %s test on numeric metadata. Aborting' % statistical_test)
```

Comment:
what about doing meandiff etc. on a metadata column with values 1,2 - will it work? (i think it should?)

Reply:
I guess it will work, but "1, 2" here are just indices of categories, not actual values, so maybe it is not justified. @serenejiang